scanner.go

```diff
@@ -142,22 +156,22 @@ func scanBasicString(b []byte) ([]byte, []byte, error) {
 		}
 	}
 
-	return nil, nil, fmt.Errorf(`basic string not terminated by "`)
+	return nil, nil, errBasicLineNotTerminatedByQuote
 }
 
+//nolint:godox
```

Comment:
Same here. Those TODOs were here when those functions were `parse*` functions in the initial v2 code, but now there are `scan*` and `decode*` functions, with the latter handling validation.